dask/array/routines.py

```diff
@@ -628,7 +628,7 @@ def histogram(a, bins=None, range=None, normed=False, weights=None, density=None
 
     # Map the histogram to all bins
     def block_hist(x, weights=None):
-        return np.histogram(x, bins, weights=weights)[0][np.newaxis]
+        return np.histogram(x, bins, range=range, weights=weights)[0][np.newaxis]
```

Comment:
We might want to include `range` as an argument to this function and then include that argument in the task explicitly, rather than closing over it in `block_hist`.  This might make the graph a bit more explicit and interpretable, which can help in the rare case that we inspect it directly for debugging.

Otherwise this looks great to me.  Thanks @jrbourbeau !

Reply:
Good point @mrocklin, I'll go ahead and add `range` as a parameter to `block_hist`